Adds data-feed-proxy-combinators docs

acenolaza opened the pull request:
Closes https://github.com/api3dao/data-feed-proxy-combinators/issues/26

wkande commented:
Another good catch, thanks.

acenolaza committed:
Adds data-feed-proxy-combinators docs

acenolaza committed:
Adds data-feed-proxy-combinators sidebar link

Siegrift commented:
I'd just include it in https://oev-docs--pr210-data-feed-proxy-comb-40m36top.web.app/dapps/integration/api3-contracts.html I don't think we need the sidebar.

Siegrift commented on docs/dapps/integration/data-feed-proxy-combinators.md:
```diff
@@ -0,0 +1,30 @@
+
+# `data-feed-proxy-combinators`
+
+The [`data-feed-proxy-combinators`](https://github.com/api3dao/data-feed-proxy-combinators) repository provides a powerful toolkit of modular smart contracts. These "combinators" allow you to create new, customized data feeds by transforming and combining existing Api3 dAPIs or even external data feeds. This is particularly useful when your dApp requires a data feed that is not directly available ...
```
```suggestion
The [`data-feed-proxy-combinators`](https://github.com/api3dao/data-feed-proxy-combinators) repository provides a powerful toolkit of modular smart contracts. These "combinators" allow you to create new, customized data feeds by transforming and combining existing Api3 feeds or even external data feeds. This is particularly useful when your dApp requires a data feed that is not directly available on the Api3 Market.
```

we don't use dAPI in the user facing docs.

Siegrift commented on docs/dapps/integration/data-feed-proxy-combinators.md:
```diff
@@ -0,0 +1,30 @@
+
+For comprehensive instructions on how to use and deploy these contracts, including practical use-case scenarios, please refer to the [`README.md`](https://github.com/api3dao/data-feed-proxy-combinators/blob/main/README.md) file.
+
+::: info ℹ️ Source of Truth
```
I'd remove this info section.

Siegrift commented on docs/dapps/integration/data-feed-proxy-combinators.md:
```diff
@@ -0,0 +1,30 @@
+
+## Core Components
+
+The repository offers a suite of composable proxy contracts. You can chain these together to build complex data feed pipelines. The primary contracts include:
```
```suggestion
The repository offers the following suite of composable contracts:
```

Siegrift commented on docs/dapps/integration/data-feed-proxy-combinators.md:
```diff
@@ -0,0 +1,30 @@
+The repository offers a suite of composable proxy contracts. You can chain these together to build complex data feed pipelines. The primary contracts include:
+
+- **`InverseApi3ReaderProxyV1.sol`**: Computes the inverse of a data feed's value (e.g., converting ETH/USD to USD/ETH).
+- **`ProductApi3ReaderProxyV1.sol`**: Multiplies the values of two data feeds to derive a new cross-rate (e.g., `(A/B) * (B/C) = A/C`).
```
```suggestion
- **`ProductApi3ReaderProxyV1.sol`**: Multiplies the values of two data feeds to derive a new cross-rate (e.g., multiplying "wstETH/stETH Exchange Rate" with ETH/USD feed to get wstETH/USD).
```

Siegrift commented on docs/dapps/integration/data-feed-proxy-combinators.md:
```diff
@@ -0,0 +1,30 @@
+
+## Usage and Examples
+
+For comprehensive instructions on how to use and deploy these contracts, including practical use-case scenarios, please refer to the [`README.md`](https://github.com/api3dao/data-feed-proxy-combinators/blob/main/README.md) file.
```
The usage and examples are documented in the [README.md](link) of the repository.

acenolaza committed:
Applies suggestions from PR

Siegrift reviewed (changes requested):
@acenolaza not sure if you saw
```
I'd just include it in https://oev-docs--pr210-data-feed-proxy-comb-40m36top.web.app/dapps/integration/api3-contracts.html I don't think we need the sidebar.
```

acenolaza commented:
> @acenolaza not sure if you saw
> 
> ```
> I'd just include it in https://oev-docs--pr210-data-feed-proxy-comb-40m36top.web.app/dapps/integration/api3-contracts.html I don't think we need the sidebar.
> ```

Sorry, missed that message. I'll apply that change

acenolaza committed:
Removes sidebar link to data-feed-proxy-combinators.md

acenolaza commented:
> I'd just include it in https://oev-docs--pr210-data-feed-proxy-comb-40m36top.web.app/dapps/integration/api3-contracts.html I don't think we need the sidebar.

Removed the sidebar link, added a short description and link to api3-contracts.md but I'm wondering if [here](https://oev-docs--pr210-data-feed-proxy-comb-40m36top.web.app/dapps/integration/contract-integration.html#reading-the-data-feed) wouldn't be a better place to like the data-feed-proxy-combinators repo 🤔

Siegrift commented:
> I'm wondering if [here](https://oev-docs--pr210-data-feed-proxy-comb-40m36top.web.app/dapps/integration/contract-integration.html#reading-the-data-feed) wouldn't be a better place to like the data-feed-proxy-combinators repo 🤔

Yeah, maybe it's better to mention it after the mixed oracle design section.

acenolaza committed:
Moves data-feed-proxy-combintors.md link from api3-contracts-md to contract-integration.md

acenolaza commented:
> Yeah, maybe it's better to mention it after the mixed oracle design section.

Ended up placing the text **before** the mixed oracle design section and right after the reading a data feed section

Siegrift commented on docs/dapps/integration/contract-integration.md:
```diff
@@ -140,6 +140,10 @@ We recommend directing them to this page.
 
+## Combining Data Feeds
+
+The `data-feed-proxy-combinators` repository offers modular smart contracts for creating customized data feeds. These can be formed by transforming and combining existing feeds, which is useful when a specific data feed is not available on the Api3 Market. This allows dApps to integrate highly specific or composite data requirements directly into their smart contracts. See [here.](/dapps/integrat ...
```
```suggestion
which is useful when a specific data feed is not available on the Api3 Market
```

is confusing. I'd just remove this part of sentence.

acenolaza replied:
👍🏻 
~Where should I put the link to `/dapps/integration/data-feed-proxy-combinators.md` ?~
Nvm, go it now.

Siegrift commented on docs/dapps/integration/contract-integration.md:
```diff
@@ -140,6 +140,10 @@ We recommend directing them to this page.
 
 :::
 
+## Combining Data Feeds
```
Burak doesn't use the capitalized titles on the docs (apart from service names and coined terms) so this should be just "Combining data feeds"

Siegrift reviewed (changes requested):
Final changes from me.

acenolaza committed:
Minor adjustments to Combining data feeds

Siegrift commented on docs/dapps/integration/contract-integration.md:
```diff
@@ -140,6 +140,10 @@ We recommend directing them to this page.
 
+## Combining data feeds
+
+The `data-feed-proxy-combinators` repository offers modular smart contracts for creating customized data feeds. These can be formed by transforming and combining existing feeds. This allows dApps to integrate highly specific or composite data requirements directly into their smart contracts. See [here.](/dapps/integration/data-feed-proxy-combinators.md) for more details.
```
```suggestion
The `data-feed-proxy-combinators` repository offers modular smart contracts for creating customized data feeds. These can be formed by transforming and combining existing feeds. This allows dApps to integrate highly specific or composite data requirements directly into their smart contracts. See [here](/dapps/integration/data-feed-proxy-combinators.md) for more details.
```

Siegrift replied:
But thinking about the final changes, I dislike a page that is only navigatable by a link and not available on the sidebar. I think let's just include the link to the [README](https://github.com/api3dao/data-feed-proxy-combinators/blob/main/README.md) here and remove the `docs/dapps/integration/data-feed-proxy-combinators.md`

Siegrift reviewed (approved):
I've changed my mind on how I like the looks of it, but approving. Once you fix this request review from Burak as he's responsible for the dApp's part of docs.

acenolaza committed:
Removes data-feed-proxy-combinators.md and redirects to README.md

bbenligiray commented on docs/dapps/integration/contract-integration.md:
```diff
@@ -140,6 +140,10 @@ We recommend directing them to this page.
+## Combining data feeds
+
+The `data-feed-proxy-combinators` repository offers modular smart contracts for creating customized data feeds. These can be formed by transforming and combining existing feeds. This allows dApps to integrate highly specific or composite data requirements directly into their smart contracts. See [here](https://github.com/api3dao/data-feed-proxy-combinators/blob/0beed2baa94a48b50841f4a5ae6aeaccfe4 ...
+
```
```suggestion
## Api3ReaderProxyV1 combinations

See the [`data-feed-proxy-combinators`](https://github.com/api3dao/data-feed-proxy-combinators) repository for various modular contracts that you can use to create combinations out of Api3ReaderProxyV1 contracts.
For example, you can combine an `ETH/USD` Api3ReaderProxyV1 contract and a `wstETH/ETH Exchange Rate` Api3ReaderProxyV1 contract to read a `wstETH/USD` value.
```

bbenligiray replied:
I guess I should ask for a review back

bbenligiray replied:
My assumption is that the data feeds to be combined are all read from Api3ReaderProxyV1 contracts, in which case it's better to be specific. Otherwise it may create confusion with the following mixed oracle design section.

acenolaza replied:
Yes, there's just [one other contract](https://github.com/api3dao/data-feed-proxy-combinators/blob/24-proxy-dappid/contracts/adapters/ScaledApi3FeedProxyV1.sol#L0-L1) that enables reading a Chainlink feed and exposes the Api3ReaderProxyV1

bbenligiray replied:
That doesn't look inconsistent with the suggestion to me, unless you mean that "combination" implies multiple Api3ReaderProxyV1 contracts and that uses a single one

acenolaza replied:
Nope, and that is why that contract is in an "adapters" folder meaning that it is only supposed to be used in special cases in order to adapt a Chainlink feed to an Api3ReaderProxyV1 before combining it 👍🏻

acenolaza committed:
Update docs/dapps/integration/contract-integration.md

Co-authored-by: Burak Benligiray <bbenligiray@gmail.com>

acenolaza committed:
Format
